src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -54,16 +54,13 @@ export const monoPlatform: Platform = {
     });
   },
 
-  callEntryPoint: function callEntryPoint(assemblyName: string) {
-    // Instead of using Module.mono_call_assembly_entry_point, we have our own logic for invoking
-    // the entrypoint which adds support for async main.
+  callEntryPoint: function callEntryPoint(assemblyName: string) : Promise<any> {
     // Currently we disregard the return value from the entrypoint, whether it's sync or async.
     // In the future, we might want Blazor.start to return a Promise<Promise<value>>, where the
     // outer promise reflects the startup process, and the inner one reflects the possibly-async
     // .NET entrypoint method.
-    const invokeEntrypoint = bindStaticMethod('Microsoft.AspNetCore.Components.WebAssembly', 'Microsoft.AspNetCore.Components.WebAssembly.Hosting.EntrypointInvoker', 'InvokeEntrypoint');
-    // Note we're passing in null because passing arrays is problematic until https://github.com/mono/mono/issues/18245 is resolved.
-    invokeEntrypoint(assemblyName, null);
+    const emptyArray = [ [ ] ];
+    return BINDING.call_assembly_entry_point(assemblyName, emptyArray, undefined);
```

Comment:
The `emptyArray` is the part I was unsure about given the original `EntrypointInvoker` set the args [based on the entrypoint parameters](https://github.com/dotnet/aspnetcore/pull/31769/files#diff-fb97fa9739af7e185d6cce1478623749e70827a919ebb889efe0f94519628f04L27). Does anyone know if this is still required?

Reply:
There's also a `MONO.mono_call_assembly_entry_point` API. Maybe that's preferable to the BINDING API? @lewing ?

Reply:
I'd copy what is in the link, so pass "m" as the last arg and skip the signature lookup. `[[]]` should be fine in any case but doing as runtime-test.js does will allow `[["arg0", "arg1"]]` to actually pass to `Main(string[])`

Reply:
Thanks, updating with:
```suggestion
    const emptyArray = [ [ ] ];
    return BINDING.call_assembly_entry_point(assemblyName, emptyArray, "m");
```

Just to confirm, the second arg should be left as is, or are you suggesting to change it somehow as well?

Reply:
That should be safe for all cases. You would only need to change it if you wanted to pass actual arguments to Main.